models/evc.py

```diff
@@ -1201,51 +1198,34 @@ def check_list_traces(list_circuits):
         if not list_circuits:
             return {}
         uni_list = []
-        circuit_data = {}
-        for circuit_id in list_circuits:
-            circuit = list_circuits[circuit_id]
+        for circuit in list_circuits:
             # if a inter-switch EVC does not have current_path, it does not
             # make sense to run sdntrace on it
             if not circuit.is_intra_switch() and not circuit.current_path:
+                list_circuits.remove(circuit)
```

Comment:
This logic here is way too late and making it harder to reason and test. This conditional can be moved to the pre-filtering that takes place https://github.com/kytos-ng/mef_eline/blob/fix_branch/main.py#L91-L98 when buliding the `list_circuits`, could you refactor this?


In the future, not needed in this PR, we can also extract all of those conditionals as a function for better testability something like `should_be_checked(circuit)`.

Reply:
Sure, I agree.

Reply:
Looks really great. Let me close this thread.